src/components/CmsIntersectionObserver.vue

```diff
@@ -0,0 +1,36 @@
+<template>
+  <div class="observer" />
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+
+@Component({ name: 'cms-intersection-observer' })
+export default class CmsIntersectionObserver extends Vue {
+  @Prop({ type: Object, default: () => {} }) options!: object;
+
+  observer: IntersectionObserver | null = null;
+
+  mounted() {
+    this.observer = new IntersectionObserver(([entry]) => {
+      if (entry) {
+        this.$emit(entry.isIntersecting ? 'enter' : 'leave');
+      }
+    }, this.options);
+    this.observer.observe(this.$el);
+  }
+
+  destroyed() {
+    if (this.observer) {
+      this.observer.disconnect();
+    }
+  }
+}
+</script>
+
+<style lang="less" scoped>
```

Comment:
is it easy to use scss here?

Reply:
Yeah, I think so, but the other components (CmsInspectSheet and CmsZone use less at the moment). Thoughts?

Reply:
Whoops. Forgot there were other components with scss. I'll follow up with another PR.